scipy/stats/_qmc.py

```diff
@@ -147,7 +171,11 @@ def scale(sample, l_bounds, u_bounds, reverse=False):
         return (sample - lower) / (upper - lower)
 
 
-def discrepancy(sample, iterative=False, method="CD", workers=1):
+def discrepancy(
+        sample: npt.ArrayLike,
+        iterative: bool = False,
+        method: str = "CD",
+        workers: int = 1) -> DecimalNumber:
```

Comment:
Same comment as above: try to avoid returning unions.
Based on the annotations in https://github.com/scipy/scipy/blob/master/scipy/stats/_qmc_cy.pyi it seems to return a normal `float`.

```suggestion
        workers: int = 1) -> float:
```